### ISSUE TITLE
False positive with OptionalAbstractKeyword

### ISSUE THREAD

**Issue by tasomaniac** [arturbosch/detekt#879]
## Expected Behavior
I have a valid code (pasted below) that shouldn't have this warning. There is no other way of doing this one actually. 

## Current Behavior
`OptionalAbstractKeyword` is warning given below. The `abstract` keyword is not even in an `interface` but in another `abstract class` that is inside an `interface`.

![image](https://user-images.githubusercontent.com/763339/39211042-7912c31a-47d8-11e8-957e-a4afb2da2526.png)

## Steps to Reproduce (for bugs)

The code looks like below. It is a `Dagger` component.

`AndroidInjector.Builder` is abstract.
`@Component.Builder` is required to be put on an `abstract` class.
`fun build(): AppComponent` is abstract. It is defined in `AndroidInjector.Builder` actually. It overrides and modifies the return type.

```kotlin
@Component
interface AppComponent : AndroidInjector<App> {

    fun analytics(): Analytics

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>() {

        abstract override fun build(): AppComponent
    }
}
```

## Context
I don't believe I have an error to fix. But other than that the message shown above does not have much information and does not give any additional information about how to fix the issue. 

## Your Environment
* Version used: 1.0.0.RC6-3
* Operating System and version: Mac OS
* Link to your project: https://github.com/tasomaniac/OpenLinkWith

**Comment by schalkms** [comment]
I tried to reproduce your error without success. I used your code from the github link (nice project btw). Are you using the latest version of detekt?
The issue message could be improved I guess. It should tell the user about unnecessary abstract modifiers in an interface.

**Comment by arturbosch** [comment]
We had some false positives with that rule, I think they should be fixed in RC6.4 or master.
@schalkms he is using RC6.4.

**Comment by schalkms** [comment]
I overlooked it. He is using RC6-3 actually ;)

**Comment by tasomaniac** [comment]
Yes, there is a small blocker preventing me to upgrade at the moment. I searched for old issues but couldn't find anything. I will double check.

**Comment by skimarxall** [comment]
Hi, I am having an issue with this rule with sealed classes.

Unless I am mistaken this is okay to do in sealed classes. 
```kotlin
sealed class Error {

        abstract val code: Int

        data class Cancelled(override val code: Int = -1) : Error()
        data class Network(override val code: Int = -1) : Error()
        data class Internal(override val code: Int = 500) : Error()
}
```
I am using the latest RC6-4. An one important remarkt this class is inside an interface.

**Comment by skimarxall** [comment]
Ok, the issue is because the sealed class is inside an interface. If I move it out, it does not complain.

**Comment by schalkms** [comment]
Do we have any update on this issue?
